server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -1104,22 +1076,11 @@ func (e *EventChannel) collectAPIFacets(iid string, event *build_event_stream.Bu
 
 func (e *EventChannel) writeBuildMetadata(ctx context.Context, invocationID string) error {
 	db := e.env.GetInvocationDB()
-	ti := &tables.Invocation{
-		InvocationID: invocationID,
-	}
-	invocationProto := TableInvocationToProto(ti)
-	err := e.fillInvocationFromEvents(
-		ctx,
-		GetStreamIdFromInvocationIdAndAttempt(invocationID, e.attempt),
-		invocationProto,
-	)
-	if err != nil {
-		return err
-	}
+	invocationProto := e.parser.GetInvocation()
 	if e.logWriter != nil {
 		invocationProto.LastChunkId = e.logWriter.GetLastChunkId(ctx)
 	}
-	ti, err = e.tableInvocationFromProto(invocationProto, ti.BlobID)
+	ti, err := e.tableInvocationFromProto(invocationProto, "" /*=blobID*/)
```

Comment:
`ti.BlobID` was always `""` here in the previous code, because we were creating a `tables.Invocation{}` without ever setting `BlobID`. I'm not sure if it that was intended or not, but the new code preserves the existing behavior.

I guess it doesn't matter too much because we set BlobID to the invocation ID in the `FinalizeInvocation` call.